### DISCUSSION TITLE
How to create another relationship from the data I already have in the database

### DISCUSSION THREAD

**Discussion by koksikus** [directus/directus#15359] [Archive]
I have a collection where I identify data by several digit numbers that are unique for each record. However, there are situations where this number is missing ... So the key in this collection is the ID. 
Now I have an idea to create another collection in which the key will be This number and the next data will be assigned in the record. Now, when I am trying to create a M2O or O2M relation, in relations in the collection where there is an ID, I cannot select a field with this number, because the field is without editing, only the ID is shown. In this new field I have to put this number again to show one of the data in the new table.
To sum up, the idea is to create a new field in the collection, but in it select relations to fields that exist and display only the result of the relation in this field ...
I know the easiest way is to add a field to an existing collection and enter the data there and then update it. However, this data comes from a different database and it is easier to remove adding data to a new collection because views are still not supported.

Hope I made myself clear :) thank you in advance

**Comment by rijkvanzanten** [comment]
It sounds like you're asking to support foreign key constraints to non-primary key fields (https://github.com/directus/directus/discussions/8720). That's currently not supported yet

**Reply by koksikus** [reply]
This is exactly what I was looking for.  Another nice option to be able to include more data in collections.  Thanks for the answer.
